stream/build.gradle.kts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2022-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+    id("com.hedera.block.protobuf")
+    alias(libs.plugins.pbj)
+}
+
+group = "com.hedera.block"
+
+description = "Hedera API"
+
+// Remove the following line to enable all 'javac' lint checks that we have turned on by default
+// and then fix the reported issues.
+tasks.withType<JavaCompile>().configureEach {
+    options.compilerArgs.add("-Xlint:-exports,-deprecation,-removal")
+}
+
+// Add downloaded HAPI repo protobuf files into build directory and add to sources to build them
+tasks.cloneHederaProtobufs {
+    // uncomment below to use a specific tag
+    // tag = "v0.53.0"
+    // uncomment below to use a specific branch
+    branch = "main"
+}
+
+sourceSets {
+    main {
+        pbj {
+            srcDir(tasks.cloneHederaProtobufs.flatMap { it.localCloneDirectory.dir("services") })
+            srcDir(tasks.cloneHederaProtobufs.flatMap { it.localCloneDirectory.dir("block") })
+            srcDir(tasks.cloneHederaProtobufs.flatMap { it.localCloneDirectory.dir("platform") })
+            srcDir(tasks.cloneHederaProtobufs.flatMap { it.localCloneDirectory.dir("streams") })
+        }
+        proto {
+            srcDir(tasks.cloneHederaProtobufs.flatMap { it.localCloneDirectory.dir("services") })
+            srcDir(tasks.cloneHederaProtobufs.flatMap { it.localCloneDirectory.dir("block") })
+            srcDir(tasks.cloneHederaProtobufs.flatMap { it.localCloneDirectory.dir("platform") })
+            srcDir(tasks.cloneHederaProtobufs.flatMap { it.localCloneDirectory.dir("streams") })
+        }
+    }
+}
+
+dependencies {
+
+    // requires transitive in the module-info.java
+    api("com.google.protobuf:protobuf-java:3.24.0") { because("com.google.protobuf") }
+    api("com.hedera.pbj:pbj-runtime:0.9.2") { because("com.hedera.pbj.runtime") }
+    api("io.grpc:grpc-stub:1.64.0") { because("io.grpc.stub") }
+    api("io.grpc:grpc-api:1.64.0") { because("io.grpc") }
+    api("com.google.guava:guava:33.0.0-jre") { because("com.google.common") }
+
+    // requires static in the module-info.java
+    runtimeOnly("com.github.spotbugs:spotbugs-annotations:4.7.3") {
+        because("com.github.spotbugs.annotations")
+    }
+    runtimeOnly("javax.annotation:javax.annotation-api:1.3.2") { because("java.annotation") }
+}
+
```

Comment:
```suggestion
```

`dependencies {}` should not be used, as the dependencies are derived rom `module-info.java`. See also my comment on `server/build.gradle.kts`.

Reply:
I spent a lot of time banging my head on this one. I saw that the `hedera-services` code didn't have these dependencies but I kept shifting and getting different compiler errors. It looks like the key is the last line you suggested in the `com.hedera.block.jpms-modules.gradle.kts` file. Thanks taking a look at this and pointing me toward a better practice! Fixed.

Reply:
Understandable. I struggled with this as well. It's a legacy of the `javax.annotation` package that is not compatible with Java Modules here.

I created an issue for me to investigate how we can get rid of this: https://github.com/hashgraph/hedera-services/issues/15077